synapse/lib/types.py

```diff
@@ -1233,7 +1232,10 @@ class Data(Type):
     stortype = s_layer.STOR_TYPE_MSGP
 
     def norm(self, valu):
-        s_common.reqjsonsafe(valu)
+        try:
+            s_common.reqjsonsafe(valu)
+        except s_exc.MustBeJsonSafe as e:
+            raise s_exc.BadTypeValu() from e
```

Comment:
mesg = ... ?

Reply:
and type=self.type.full ? we have some common bits that help people understand what form failed to norm a thing.

Reply:
``name=self.name``